src/ess/reflectometry/calibrations.py

```diff
@@ -31,12 +31,14 @@ def calibration_factor(
     :
         Calibration factor at the midpoint of each Q-bin.
     """
-    q = sc.midpoints(qbins)
+    q = da.coords['Q']
```

Comment:
It's because the supermirror correction is applied in the $\lambda, z$ coordinates.
The supermirror correction is a function of the reference $Q$, that's why we can't evaluate it on `qbins`, they refer to the sample $Q$.

I agree it's not very elegant to get the entire data array just to extract the $Q$ coordinate.

Another option would be to apply the supermirror efficiency correction directly on the reference events, similar to the footprint correction. However, it's less efficient and I don't think it would impact the result.